packages/@react-spectrum/datepicker/src/TimeField.tsx

```diff
@@ -44,6 +44,17 @@ function TimeField<T extends TimeValue>(props: SpectrumTimeFieldProps<T>, ref: F
   let inputRef = useRef(null);
   let {labelProps, fieldProps, descriptionProps, errorMessageProps} = useTimeField(props, state, inputRef);
 
+  /**
+   * When a TimeField is in a Picker, the Picker can close without triggering
+   * a TimeField onBlur which is sometimes used to persist valid dates. This
+   * persists the value the second it is valid.
+   */
+  useEffect(() => {
+    return () => {
+      state.confirmPlaceholder();
+    };
+  }, [state]);
```

Comment:
I guess this would make it run every time `state` changes, but I we only really want it to run if the component is actually unmounting. Another problem is that in React strict mode, this cleanup will be run on the initial mount as well since effects run twice. That makes it harder to detect a "real" unmount. One way I found is something like this:

```js
let ref = useRef();
useEffect(() => {
  return () => {
    if (ref.current == null) {
      // Actual unmount rather than state change or extra strict mode effect.
    }
  };
}, [state, ref]);
```

If the DOM ref has been detached already we know it's a real unmount rather than a state change. Sandbox to demonstrate: https://codesandbox.io/s/determined-panka-chl6zn?file=/src/App.js:566-713

You could also move this into `useDateField` (which is used by `useTimeField`) so it also works in our hooks examples.

Reply:
I kept this in RSP because I was concerned about it being a breaking change for the hooks.
Thanks for the code suggestion!

Reply:
I left a comment about how this didn't work as expected so a tried something else. It doesn't work when you use the spin button arrow up or down once (fix is in progress).

Reply:
I tried this using useLayoutEffect in useDateField.js and it still threw two onChange events like the code being commented on. As mentioned in a previous comment, when this is done with useEffect we get the original behavior. It also has the bug of not being able to delete the am/pm segment with backspace mentioned in the refactor related comments.

Interesting note, if you set a time like "9:09" the first onChange has "9:00" while the second has "9:09". If you change the granularity to seconds and set the time to "9:09:09", the first onChange is "9:09:00" and the second is "9:09:09".